internal/shard/shard_manager.go

```diff
@@ -57,7 +61,6 @@ func (manager *ShardManager) Run(ctx context.Context) {
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		manager.listenForErrors()
 	}()
```

Comment:
I think we can clean up this goroutine, it's simply adding to wait group and marking it done